fdbserver/VersionedBTree.actor.cpp

```diff
@@ -3798,15 +3813,22 @@ class DWALPager final : public IPager2 {
 		// Wait for outstanding commit.
 		wait(self->commitFuture);
 
-		// While the remap queue isn't empty, advance the commit version and oldest readable version
-		// by the remap cleanup window and commit
-		while (self->remapQueue.numEntries > 0) {
+		// Set remap cleanup window to 0 to allow the remap queue to drain.
+		state int64_t remapCleanupWindowBytes = self->remapCleanupWindowBytes;
+		self->remapCleanupWindowBytes = 0;
+
+		// Try twice to commit and advance version. The first commit should trigger a remap cleanup actor, which picks
+		// up the new remap cleanup window being 0. The second commit waits for the remap cleanup actor to finish.
+		state int attempt = 0;
+		for (attempt = 0; attempt < 2; attempt++) {
 			self->setOldestReadableVersion(self->getLastCommittedVersion());
-			wait(self->commit(self->getLastCommittedVersion() + self->remapCleanupWindow + 1));
+			wait(self->commit(self->getLastCommittedVersion() + 1));
 		}
+		ASSERT(self->remapQueue.numEntries == 0);
 
-		// One final commit because the active commit cycle may have popped from the remap queue
-		wait(self->commit(self->getLastCommittedVersion() + 1));
+		// Restore remap cleanup window.
+		if (remapCleanupWindowBytes != 0)
```

Comment:
Since the goal here is just to restore whatever the prior value was, the check is not necessary.